frontend/src/components/forms/TextInput.tsx

```diff
@@ -11,16 +14,54 @@ type Props = InputHTMLAttributes<HTMLInputElement> & {
   /**
    * The register function from `react-hook-form`'s `useForm()` hook used for validation and submission
    */
-  register?: FormRegisterType
+  register?: UseFormRegister<any>
+  /**
+   * A set of options in the format supported by `react-hook-form`.
+   *
+   * ⚠️ If validations for `min`, `max`, `minLength`, `maxLength`, or `required`
+   * are provided in this object, they will override the props of the same name
+   * passed individually. This allows users to define custom error messages if
+   * they so wish:
+   *
+   * @example
+   * <TextInput
+   *   label="Age"
+   *   required
+   *   min={16} />
+   *
+   * @example
+   * <TextInput
+   *   label="Age"
+   *   registerOptions={{
+   *     required: "Age is a required field",
+   *     min: {
+   *       value: 16,
+   *       message: "You must be at least 16 to create offers"
+   *     }
+   *   }} />
+   */
```

Comment:
@bion does this answer your question about `otherProps` vs `registerOptions`?

Reply:
Yep, thanks!